### PR TITLE
radio component

### PR DESCRIPTION
**Pull request by 1297947244**

### CODE REVIEW

File: docs/src/components/fields/RadioGroupField.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { RadioGroupTextField, RadioGroupTextBuilder } from './RadioGroupTextField';
+import KeyValueDisplay from './KeyValueDisplay';
+import RadioGroupOptionEditor from './RadioGroupOptionEditor';
+
+const Type = "Teamious.radioButton";
+const Field: IField = {
+    id: '',
+    label: '单选按钮',
+    type: Type,
+    fields: [
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupField.ts]: 这个不应该放在fields里，fields是给嵌套用的。

这个应该放在options里

File: docs/src/components/fields/RadioGroupField.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            id: '001',
+            label: 'JAVA',
+            type: 'ABCDEFG',
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupField.ts]: What's this type used for?

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import { FormControl, Checkbox } from 'react-bootstrap';
+import * as assign from 'object-assign';
+
+import { IFieldOptionEditorProps } from 'react-dynamic-formbuilder';
+
+interface IState { }
+
+
+export default class RadioGroupOptionEditor extends React.PureComponent<IFieldOptionEditorProps, IState> {
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx]: If there is no state, you can use 'void'.

=> React.PureComponent<IFieldOptionEditorProps, void>

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    render() {
+        const { label, fields } = this.props.field;
+        const { required, unique, others } = this.props.field.options;
+        return (
+            <div>
+                <div>
+                    <span>Label</span>
+                    <FormControl type='text' value={label} onChange={this.onLabelChange} />
+                    <span>选项设置</span>
+                    {fields.map((item) => {
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx]: Why not use <OrderedListInput /> component?

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        });
+        this.props.onChange(field);
+    }
+
+    private addOption() {
+        let field = assign({}, this.props.field);
+        let new_option = {
+            id: (this.addOptionId++).toString(),
+            label: "选项"+this.addOptionId,
+            type: "ABCDEFG"
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx]: Why did you hardcode?

File: docs/src/components/fields/RadioGroupTextField.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                        {fields.map((item)=>{
+                            //TODO: (xiongchao) radio name is wrong
+                            return <Radio name={item.type} key={item.id}>{item.label}</Radio>
+                        })}
+                        {others.checked &&
+                            <div>
+                                <Radio>{others.label}</Radio>
+                                <FormControl
+                                    type='text'
+                                    value={others.value}
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupTextField.tsx]: This cannot work. field.option is form template definition. It cannot be saved into each data, and it won't change unless someone use option editor to modify.

The value in the props is the only information which will be saved in the database. Value can be a jObject if necessary.

**Review by linweigao — CHANGES_REQUESTED** [review]

File: docs/src/components/fields/RadioGroupField.ts (hunk 1 above)
Comment (1297947244) [reply docs/src/components/fields/RadioGroupField.ts]: Done.

File: docs/src/components/fields/RadioGroupField.ts (hunk 2 above)
Comment (1297947244) [reply docs/src/components/fields/RadioGroupField.ts]: type是为了表示每组radio的名字， 修改为name。

File: docs/src/components/fields/RadioGroupOptionEditor.tsx (hunk 3 above)
Comment (1297947244) [reply docs/src/components/fields/RadioGroupOptionEditor.tsx]: Done.

File: docs/src/components/fields/RadioGroupOptionEditor.tsx (hunk 4 above)
Comment (1297947244) [reply docs/src/components/fields/RadioGroupOptionEditor.tsx]: 删除盒子组件将单独做一个组件提交，不在radio中实现。

File: docs/src/components/fields/RadioGroupOptionEditor.tsx (hunk 5 above)
Comment (1297947244) [reply docs/src/components/fields/RadioGroupOptionEditor.tsx]: Done.

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 7]
```diff
@@ -91,32 +89,33 @@ export default class RadioGroupOptionEditor extends React.PureComponent<IFieldOp
...
     }
 
     private addOption() {
         let field = assign({}, this.props.field);
+        let labelArr = field.options.labels;
         let new_option = {
-            id: (this.addOptionId++).toString(),
-            label: "选项"+this.addOptionId,
-            type: "ABCDEFG"
+            id: labelArr[labelArr.length - 1].id + this.addOptionId++,
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx]: what's the logic for the id generator?
1,2,4,7,11?

Comment (1297947244) [reply docs/src/components/fields/RadioGroupOptionEditor.tsx]: 原想法是  新增加的选项id，根据最后一个选项的id拼接0，1，2，3...
现已删除 create，remove 逻辑，使用OrderedListInput component。

File: docs/src/components/fields/RadioGroupOptionEditor.tsx (hunk 4 above)
Comment (1297947244) [reply docs/src/components/fields/RadioGroupOptionEditor.tsx]: 已使用OrderedListInput component

File: docs/src/components/fields/SingleSelectorField.ts [hunk 8]
```diff
@@ -11,6 +11,7 @@ const Field: IField = {
     options: {
         hint: 'Please enter your hint',
         required: true,
+        selectOpts: ['a', 'b', 'c'],
```
Comment (1297947244) [inline docs/src/components/fields/SingleSelectorField.ts:14]: Repair the demo drop-down box error.

File: docs/src/components/App.scss [hunk 9]
```diff
@@ -94,4 +94,33 @@
  }
  .modal-container .modal, .modal-container .modal-backdrop {
    position: absolute;
+ }
+
+ .dotted-box {
```
Comment (linweigao) [inline docs/src/components/App.scss]: Where do you need this css?

**Review by linweigao — APPROVED** [review]

File: docs/src/components/App.scss (hunk 9 above)
Comment (1297947244) [reply docs/src/components/App.scss]: Sorry, I forgot to delete it.

**Pull request merged**